Automattic/jetpack
# Reorg: Experimental: Jetpack development CLI

This is a pet project during the holidays. I find myself having to refer to `package.json` too often to remember the various scripts that we have set there—not to mention the various build and deploy scripts that aren't listed there.

We could have a much better dev experience by having a proper CLI.

This is not ready to be merged into the reorg, but PRing it for early feedback.

#### Changes proposed in this Pull Request:
* Adds a development CLI.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
n/a

#### Testing instructions:
* Run `yarn cli-init`. This will `yarn install` and both unlink/link the bins.
* Try `jetpack` at the command line at the repo root. It won't really do anything.
* Try `jetpack -h` to see the help text.
* Try `jetpack build` to have a wizard guide you to pick a project (jetpack will build, everything else will say no build step has been indicated).
* Try `jetpack build -d` (for "default" which is a shortcut to saying Jetpack). This will build JP.
* Try `jetpack build plugins/jetpack` This will build JP.

#### Proposed changelog entry for your changes:
* n/a
